pkg/test/framework/components/echo/api.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package echo
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	envoy_admin_v2alpha "github.com/envoyproxy/go-control-plane/envoy/admin/v2alpha"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/test/application/echo"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+// CallProtocol enumerates the protocol options for calling an Endpoint endpoint.
+type CallProtocol string
+
+const (
+	// HTTP calls echo with HTTP
+	HTTP CallProtocol = "http"
+
+	// GRPC calls echo with GRPC
+	GRPC CallProtocol = "grpc"
+
+	// WebSocket calls echo with WebSocket
+	WebSocket CallProtocol = "ws"
+)
+
+// Instance is a component that provides access to the deployed echo service.
+type Instance interface {
```

Comment:
This interface look similar to the one defined in [apps.go](https://github.com/istio/istio/blob/4414da92c00edf9e30b76059051c036ed26fd008/pkg/test/framework/components/apps/apps.go#L54). Wouldn't it be nice to reuse some stuff?

Reply:
This is going to replace apps.